src/pool/reserve.ts

```diff
@@ -189,75 +328,38 @@ export class ReserveConfig {
       reactivity
     );
   }
-
-  static ReserveConfigToXDR(reserveConfig?: ReserveConfig): xdr.ScVal {
-    if (!reserveConfig) {
-      return xdr.ScVal.scvVoid();
-    }
-    const arr = [
-      new xdr.ScMapEntry({
-        key: ((i) => xdr.ScVal.scvSymbol(i))('c_factor'),
-        val: ((i) => xdr.ScVal.scvU32(i))(reserveConfig.c_factor),
-      }),
-      new xdr.ScMapEntry({
-        key: ((i) => xdr.ScVal.scvSymbol(i))('decimals'),
-        val: ((i) => xdr.ScVal.scvU32(i))(reserveConfig.decimals),
-      }),
-      new xdr.ScMapEntry({
-        key: ((i) => xdr.ScVal.scvSymbol(i))('index'),
-        val: ((i) => xdr.ScVal.scvU32(i))(reserveConfig.index),
-      }),
-      new xdr.ScMapEntry({
-        key: ((i) => xdr.ScVal.scvSymbol(i))('l_factor'),
-        val: ((i) => xdr.ScVal.scvU32(i))(reserveConfig.l_factor),
-      }),
-      new xdr.ScMapEntry({
-        key: ((i) => xdr.ScVal.scvSymbol(i))('max_util'),
-        val: ((i) => xdr.ScVal.scvU32(i))(reserveConfig.max_util),
-      }),
-      new xdr.ScMapEntry({
-        key: ((i) => xdr.ScVal.scvSymbol(i))('r_one'),
-        val: ((i) => xdr.ScVal.scvU32(i))(reserveConfig.r_one),
-      }),
-      new xdr.ScMapEntry({
-        key: ((i) => xdr.ScVal.scvSymbol(i))('r_three'),
-        val: ((i) => xdr.ScVal.scvU32(i))(reserveConfig.r_three),
-      }),
-      new xdr.ScMapEntry({
-        key: ((i) => xdr.ScVal.scvSymbol(i))('r_two'),
-        val: ((i) => xdr.ScVal.scvU32(i))(reserveConfig.r_two),
-      }),
-      new xdr.ScMapEntry({
-        key: ((i) => xdr.ScVal.scvSymbol(i))('reactivity'),
-        val: ((i) => xdr.ScVal.scvU32(i))(reserveConfig.reactivity),
-      }),
-      new xdr.ScMapEntry({
-        key: ((i) => xdr.ScVal.scvSymbol(i))('util'),
-        val: ((i) => xdr.ScVal.scvU32(i))(reserveConfig.util),
-      }),
-    ];
-    return xdr.ScVal.scvMap(arr);
-  }
 }
 
 export class ReserveData {
```

Comment:
I'm not sure I agree with both `ReserveData` and `ReserveConfig` having both a class and interface. I'd rather stick with one or the other.

Have you checked if the encoding algorithm in the contract spec works for a class if the attribute names are spelled correctly? It seems like the function attributes could also trip the encoder up, but it's not obvious.

I'd recommend finding a way to discard one of the objects. Seems like reasonable options to me are:
1. Fix attribute naming for the class (would need to verify `pool_client` can take an object of the class as an argument to a contract call and encode it properly)
2. Leave it as an interface, and write a `ReserveConfigFromXDR` function that lives next to the interface, manually build the key when loading the higher level `Reserve` object

Reply:
Ditto for any other case I might have missed where there is an identical `class` and `interface` definition